src/cli/cli_dataset.cpp

```diff
@@ -912,7 +912,22 @@ otError Dataset::ProcessUpdater(Arg aArgs[])
     }
     else if (aArgs[0] == "start")
     {
-        error = otDatasetUpdaterRequestUpdate(mInterpreter.mInstance, &sDataset, &Dataset::HandleDatasetUpdater, this);
+        if (aArgs[1] == "tlvs")
+        {
+            MeshCoP::Dataset         dataset;
+            otOperationalDatasetTlvs datasetTlvs;
+
+            SuccessOrExit(error = dataset.SetFrom(*static_cast<MeshCoP::Dataset::Info *>(&sDataset)));
+            dataset.ConvertTo(datasetTlvs);
+
```

Comment:
When or why would the CLI user want to use the `tlvs` command? 

This sub-command is converting `sDataset` to TLV sequence which is then again converted back to struct format by the implementation. To the CLI user this behaves exactly as `update start`. 

Also, in gneral we want to avoid using core internal types and definitions in CLI (e.g., `MeshCoP::Dataset` or `SetFrom()` and `ConvertTo()`). The CLI should only see and use public OT APIs and types.